### PR TITLE
Remove indices query

### PR DESCRIPTION
**Pull request by javanna**
The indices query is deprecated since 5.0.0 (#17710). It can now be removed in master (future 6.0 version).

### CODE REVIEW

**Comment by javanna** [comment]
Although we said that this query can be removed in 6.0 as it was deprecated with 5.0, this change breaks backwards compatibility on the network layer. @s1monw what is our plan on changes like this? Can we just state that a search using this query will not work anymore against a mixed version cluster? Or do we want to somehow try and build a bw comp layer around this?

**Comment by s1monw** [comment]
@javanna I think we can't support stuff that is added in that fashion either. I think if you want to use 6 you have to stop using deprecated features before you do so.

**Comment by javanna** [comment]
sounds good @s1monw thanks.

**Comment by javanna** [comment]
@elasticmachine retest this please

**Comment by javanna** [comment]
@jpountz would you mind reviewing this?

File: docs/reference/migration/migrate_6_0/search.asciidoc [hunk 1]
```diff
@@ -10,6 +10,7 @@
   only the strings `"true"` and `"false"` will be parsed into their boolean
   counterparts. Other strings will cause an error to be thrown.
 
+* The deprecated indices query has been removed.
```
Comment (jpountz) [inline docs/reference/migration/migrate_6_0/search.asciidoc]: s/indices/`indices`/?

**Review by jpountz — APPROVED** [review]
LGTM

**Pull request merged**